client/resolve.c

```diff
@@ -384,6 +383,12 @@ TDNFPrepareSinglePkg(
                       pSack,
                       queueGoal,
                       pszPkgName);
+        if (dwError == ERROR_TDNF_ALREADY_INSTALLED)
+        {
+            dwError = TDNFMarkAutoInstalledSinglePkg(pTdnf, pszPkgName);
+            BAIL_ON_TDNF_ERROR(dwError);
+            dwError = ERROR_TDNF_ALREADY_INSTALLED;
```

Comment:
Does control reach this line? after bailing out?

Reply:
Yes. If the package is already installed, we want to mark it installed because it might have been automatically installed already, but it's also wanted on its own. `TDNFMarkAutoInstalledSinglePkg` might return another error though, so we catch that first. After it's done successfully (and set `dwError` to `0`), we still want to return the original error.